### ISSUE TITLE
BUSI Usage Inquiry

### ISSUE THREAD

**Issue by MennaMagdie** [huangjin520/EMGANet#6]
Hi, I want to train your model on BUSI dataset, could you please tell me in what structure the dataset should be? and is there any other thing that should be done beside running the train BUSI script?
Thanks.

**Comment by huangjin520** [comment]
Hi, thanks for your interest!

[train_BUSI.py](https://github.com/huangjin520/EMGANet/blob/main/med_seg/train_BUSI.py)



For training on the BUSI dataset, please organize your data in the following structure:

> EMGANet/Dataset/BUSI/train/img/*.bmp
>EMGANet/Dataset/BUSI/train/mask/*.bmp

> EMGANet/Dataset/BUSI/val/img/*.bmp
>EMGANet/Dataset/BUSI/val/mask/*.bmp

> EMGANet/Dataset/BUSI/test/img/*.bmp
>EMGANet/Dataset/BUSI/test/mask/*.bmp

In the code, we simply use pandas + glob to traverse the image and mask file names and then build a dataframe like this:
```
train_data = pd.DataFrame({
    'images': sorted(glob(os.path.join(train_path, "img", "*.bmp"))),
    'masks':  sorted(glob(os.path.join(train_path, "mask", "*.bmp")))
})

```
Finally, the data interface is wrapped by our custom CrackData class:

```
train_dataset = CrackData(df=train_data, transforms=train_transforms)
```

> The above code can be used as a reference; you can also design your own data interface if preferred.

**Comment by MennaMagdie** [comment]
So if I'm using the BUSI dataset from kaggle (which are in jpg format), I need to convert them to bmp before training right?

**Comment by huangjin520** [comment]
You don’t need to convert the BUSI dataset images to .bmp. You can use them directly in .jpg format. Just adjust the code to look for .jpg files:
```
train_data = pd.DataFrame({
    'images': sorted(glob(os.path.join("train/img", "*.jpg"))),
    'masks': sorted(glob(os.path.join("train/mask", "*.jpg")))
})

```

**Comment by MennaMagdie** [comment]
Thank you for helping!
